lib/at_host/at_host.c

```diff
@@ -53,28 +53,60 @@ enum select_uart {
 
 static enum term_modes term_mode;
 static struct device *uart_dev;
-static int at_socket_fd = INVALID_DESCRIPTOR;
-static struct pollfd fds[1];
-static int nfds;
 static u8_t at_buf[AT_MAX_CMD_LEN];
 static size_t at_buf_len;
-static struct k_work at_cmd_send_work;
-static struct k_thread socket_thread;
-static K_THREAD_STACK_DEFINE(socket_thread_stack, THREAD_STACK_SIZE);
+static struct k_work cmd_send_work;
+static const char termination[3] = { '\0', '\r', '\n' };
 
+static inline void write_uart_string(char *str, size_t len)
+{
+	for (size_t i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, str[i]);
+	}
+}
 
-static const char termination[3] = { '\0', '\r', '\n' };
+static void response_handler(char *response)
+{
+	int len = strlen(response) + 1;
```

Comment:
You do not need to print the `\0`.

Reply:
I want to send the whole string including the terminator in the case someone is using something that relies on it on the other end.

Reply:
Ok. Same behavior as before when reading from the AT socket directly. Looks ok. have you tried to use the nRF Link Monitor application to validate this?

Reply:
Then you should do `chars + 1` to print the `\0` as well?

Reply:
With the current code a AT+CNUM call will look like:

AT+CNUM\r\n
242016000035955\r\n
OK\r\n\0

which is how the old output looked and how the output from the modem look as far as I remember.

Reply:
Response handler is only for notifications, but that is a response?

Reply:
A %CESQ notification looks like this

%CESQ: 59,2\r\n\0